packages/select/test/validation.test.js

```diff
@@ -0,0 +1,126 @@
+import { expect } from '@esm-bundle/chai';
+import { enterKeyDown, escKeyDown, fixtureSync, nextRender } from '@vaadin/testing-helpers';
+import sinon from 'sinon';
+import '@vaadin/item/vaadin-item.js';
+import '@vaadin/list-box/vaadin-list-box.js';
+import './not-animated-styles.js';
+import '../vaadin-select.js';
+import { html, render } from 'lit';
+
+describe('validation', () => {
+  let select, valueButton;
+
+  describe('basic', () => {
+    beforeEach(async () => {
+      select = fixtureSync('<vaadin-select></vaadin-select>');
+      select.renderer = (root) => {
+        render(
+          html`
+            <vaadin-list-box>
+              <vaadin-item value="v1" label="o2">Option 2</vaadin-item>
+              <vaadin-item value="v2" label="o2">Option 2</vaadin-item>
+              <vaadin-item value="">Option 3</vaadin-item>
+            </vaadin-list-box>
+          `,
+          root,
+        );
+      };
+      valueButton = select._valueButton;
+      await nextRender();
+    });
+
+    it('should pass the validation when the field is valid', () => {
+      select.validate();
+      expect(select.checkValidity()).to.be.true;
+      expect(select.invalid).to.be.false;
+    });
+
+    it('should not pass the validation when the field is required and has no value', () => {
+      select.required = true;
+
+      enterKeyDown(valueButton);
+      escKeyDown(valueButton);
+      expect(select.checkValidity()).to.be.false;
+      expect(select.invalid).to.be.true;
+    });
+
+    it('should validate when closing the overlay', () => {
+      const spy = sinon.spy(select, 'validate');
```

Comment:
Changed these tests to use `sinon.spy(select, 'validate')` instead of `select.validate = spy`.